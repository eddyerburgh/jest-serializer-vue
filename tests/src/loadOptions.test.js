const path = require('path');

const loadOptions = require('../../src/loadOptions.js');
const helpers = require('../../src/helpers.js');
const testHelpers = require('../helpers.js');

describe('loadOptions.js', () => {
  test('Vue config and package.json does not exist', () => {
    const fs = {
      existsSync: jest.fn(() => {
        return false;
      })
    };

    const options = loadOptions(fs);

    expect(fs.existsSync)
      .toHaveBeenCalledWith(path.join(process.cwd(), 'package.json'));

    expect(fs.existsSync)
      .toHaveBeenCalledWith(path.join(process.cwd(), 'vue.config.js'));

    expect(options)
      .toEqual(helpers.defaultSettings());
  });

  test('Manifest is loaded', () => {
    testHelpers.mockManifest({});

    const fs = {
      existsSync: jest.fn(() => {
        return true;
      })
    };

    const options = loadOptions(fs);

    expect(fs.existsSync)
      .toHaveBeenCalledWith(path.join(process.cwd(), 'package.json'));

    expect(fs.existsSync)
      .not.toHaveBeenCalledWith(path.join(process.cwd(), 'vue.config.js'));

    expect(options)
      .toEqual(helpers.defaultSettings());
  });
});
