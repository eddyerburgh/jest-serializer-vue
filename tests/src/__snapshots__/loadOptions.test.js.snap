// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadOptions.js Vue config does not exist 1`] = `
Object {
  "addInputValues": false,
  "attributesToClear": Array [],
  "formatting": Object {
    "indent_char": " ",
    "indent_inner_html": true,
    "indent_size": 2,
    "inline": Array [],
    "sep": "
",
    "unformatted": Array [
      "code",
      "pre",
    ],
  },
  "removeClassTest": false,
  "removeComments": false,
  "removeDataCy": false,
  "removeDataQa": false,
  "removeDataTest": true,
  "removeDataTestId": true,
  "removeDataTestid": true,
  "removeDataVId": true,
  "removeIdTest": false,
  "removeServerRendered": true,
  "stringifyObjects": false,
  "verbose": true,
}
`;
